runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -260,22 +261,51 @@ impl<T: Config> Pallet<T> {
 			parent_header,
 			mut disputes,
 		} = data;
+		sp_io::init_tracing();
+		let mut weight_metric = CounterVec::new(
+			"parachain_inherent_data_weight",
+			"Inherent data weight before and after filtering",
+			&vec!["when"],
```

Comment:
any reason to use a `vec!` and allocate htere and elsewhere? othwerwise, `s/&vec![/&[/g`
```suggestion
			&["when"],
```